src/test_l3_dsAudio.c

```diff
@@ -215,10 +224,49 @@ static void readFloat(float_t *choice)
 
 static void readString(char *choice)
 {
-    fgets(choice, 4, stdin);
+    scanf("%s", choice);
     readAndDiscardRestOfLine(stdin);
 }
 
+static void writeCallbackLog(char *logPath, const char *format, ...)
+{
+    va_list args;
+    va_start(args, format);
+
+
+    // Get the current timestamp in seconds and microseconds
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+
+    // Format the timestamp with year, month, day, time, and microseconds
+    char timestamp[64];
+    strftime(timestamp, sizeof(timestamp), "%Y-%m-%d %H:%M:%S", localtime(&tv.tv_sec));
+
+    // Append the microseconds to the timestamp
+    snprintf(&timestamp[strlen(timestamp)], sizeof(timestamp) - strlen(timestamp), ".%06ld", tv.tv_usec);
+
+    pthread_mutex_lock(&gCallbackMutex);
+
+    // Open the log file in append mode
+    FILE *fp = fopen(logPath, "a");
```

Comment:
we should open all logs during  module_open(), and then close all lots during module_exit()

Reply:
will handle in next version